oozie-to-airflow/converter/mappers.py

```diff
@@ -33,13 +34,38 @@
 from mappers.subworkflow_mapper import SubworkflowMapper
 
 
+class EndMapper(DummyMapper):
+    def after_parse_node(self, workflow: Workflow):
+        super().after_parse_node(workflow)
```

Comment:
Is this `super()` call needed? For now the BaseMapper's implementation does nothing. If it's just to be future-proof I'd remove it. Premature optimisation and so on.

Reply:
This is not premature optimization, but the behavior of one style of code. I assume that every method supporting an event should call superclasses. This prevents problems that will be very difficult to detect.  One convention to avoid problems. Deleting this call is premature optimisation. This style of code is natural for me.

Reply:
Maybe I chose the wrong phrase. It's not premature optimisation, it's more of a YAGNI: You Ain't Gonna Need It.

Reply:
This is needed because this is the principle of using this method. I agree that it is not necessary at the moment, but when a small amount of work can allow us to avoid problems in the future we should do it with. 

Creating an interface is defining the rules of using the method. The interface should be permanent and not changeable. We should think about how it is used now, but also how it can be used in the future. Otherwise, we'll have to keep refactoring.  I think that we have to build a house on hard foundations and also on hard principles, which is related to the thoughtful interface of the base classes.

I would like to point out how Google does it in the Android source code.
https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/app/Activity.java#1339
Their code does not do anything. They force calling superclass on execution to avoid problems in future. 

Skipping calling a super is not a good idea in this particular design pattern.